### PR TITLE
Feature/7 sign in api

### PR DESCRIPTION
**Pull request by dilyara4949**
Added new api sign in, which returns access token and user if  user email and password are found, else returns error message

### CODE REVIEW

File: internal/handler/auth.go [hunk 1]
```diff
@@ -60,6 +64,54 @@ func SignupHandler(authService AuthService, userService UserService, cfg config.
...
+			return
+		}
+
+		if req.Email == "" || req.Password == "" {
+			c.JSON(http.StatusBadRequest, response.Error{Error: "fields cannot be empty"})
+
+			return
+		}
+
+		user, err := userService.ValidateUser(c, req)
```
Comment (MihailGorelikov) [inline internal/handler/auth.go:84]: any reason to return pointer here?

File: internal/repository/errors/errors.go [hunk 2]
```diff
@@ -2,4 +2,7 @@ package errors
 
 import "errors"
 
-var ErrUserNotFound = errors.New("user not found")
+var (
+	ErrUserNotFound      = errors.New("user not found")
+	ErrIncorrectPassword = errors.New("password is not correct")
```
Comment (MihailGorelikov) [inline internal/repository/errors/errors.go:7]: invalid username name or password

**Review by MihailGorelikov — CHANGES_REQUESTED** [review]

File: internal/handler/response/signin.go [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package response
+
+type Signin struct {
+	AccessToken string `json:"access_token"`
+	User        User   `json:"user"`
```
[the remark this answers is not in the document]
Comment (dilyara4949) [reply internal/handler/response/signin.go:5]: was not sure, should i return user info, did it in case if front will need it

**Review by krylphi — APPROVED** [review]
LGTM

**Pull request merged**